moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp

```diff
@@ -134,13 +136,79 @@ void planning_scene_monitor::CurrentStateMonitor::clearUpdateCallbacks()
   update_callbacks_.clear();
 }
 
+bool planning_scene_monitor::CurrentStateMonitor::getIsRobotStopped(const std::string& group, double joint_velocity_tol,
+                                                                    double state_wait_time, bool& is_stopped)
+{
+  // Initially assume the robot is not stopped
+  is_stopped = false;
+
+  // Constant joint group variables
+  const auto& joint_models = robot_model_->getActiveJointModels();
+  const std::size_t num_joints = joint_models.size();
+  const auto joint_group = getRobotModel()->getJointModelGroup(group);
+
+  // Track the last joint states and check if the joint distance is valid given the velocity threshold
+  robot_state::RobotStatePtr previous_state = NULL;
+  ros::Time previous_time;
+  while (ros::ok())
+  {
+    if (waitForCompleteState(state_wait_time))
+    {
+      const std::pair<robot_state::RobotStatePtr, ros::Time> state_and_time = getCurrentStateAndTime();
+      const auto& current_state = state_and_time.first;
+      const auto& current_time = state_and_time.second;
+
+      if (previous_state)
+      {
+        if (current_state->hasVelocities())
```

Comment:
I think in the original implementation we allowed the state to not have velocities (which is the case for some simulations) and just checked for positions instead.